output-gazetteer.cpp

```diff
@@ -692,22 +692,20 @@ int output_gazetteer_t::process_way(osmid_t id, osmid_t *ndv, int ndc,
 
     /* Are we interested in this item? */
     if (places.has_data()) {
-        struct osmNode *nodev;
         int nodec;
 
+        assert(ndc > 0 && "no nodes to proccess");
```

Comment:
Then fix it correctly by discarding ways with 0 nodes. Before your change the malloc(0)'d pointer was never dereferenced, so things happened to work fine.